baremetal/metal3data_manager.go

```diff
@@ -505,8 +522,10 @@ func (m *DataManager) getAddressFromM3Pool(ctx context.Context, poolRef corev1.T
 
 	if err := m.client.Get(ctx, addressNamespacedName, ipAddress); err != nil {
 		if apierrors.IsNotFound(err) {
+			m.Log.Info("IPAddress not found, requeuing", "IPAddress", ipClaim.Status.Address.Name)
 			return addressFromPool{}, true, nil
 		}
+		m.Log.Error(err, "Unable to get IPAddress.", "IPAddress", ipClaim.Status.Address.Name)
```

Comment:
Not strictly related, but I added these while debugging the tests and found a test that didn't take the expected path as a consequence.